pkg/ctl/gitops/profile.go

```diff
@@ -0,0 +1,74 @@
+package gitops
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+// Command creates `generate` commands
+func Command(flagGrouping *cmdutils.FlagGrouping) *cobra.Command {
+	verbCmd := cmdutils.NewVerbCmd("generate", "Generate GitOps manifests", "")
+	cmdutils.AddResourceCmd(flagGrouping, verbCmd, generateProfileCmd)
+	return verbCmd
+}
+
+type options struct {
+	gitops.GitOptions
+	ProfilePath string
+}
+
+func generateProfileCmd(rc *cmdutils.ResourceCmd) {
+	cfg := api.NewClusterConfig()
+	rc.ClusterConfig = cfg
+
+	rc.SetDescription("profile", "Generate a GitOps profile", "")
+
+	var o options
+
+	rc.SetRunFuncWithNameArg(func() error {
+		return doGenerateProfile(rc, o)
+	})
+
+	rc.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&o.URL, "git-url", "", "", "Git repository URL")
+		fs.StringVarP(&o.Branch, "git-branch", "", "master", "Git branch; defaults to master")
+		fs.StringVarP(&o.ProfilePath, "profile-path", "", "", "Path to generate the profile in; defaults to CWD")
+		cobra.MarkFlagRequired(fs, "git-url")
+
+		cmdutils.AddNameFlag(fs, cfg.Metadata)
+		cmdutils.AddRegionFlag(fs, rc.ProviderConfig)
+		cmdutils.AddConfigFileFlag(fs, &rc.ClusterConfigFile)
+	})
+
+	cmdutils.AddCommonFlagsForAWS(rc.FlagSetGroup, rc.ProviderConfig, false)
+}
+
+func doGenerateProfile(rc *cmdutils.ResourceCmd, o options) error {
+	if err := cmdutils.NewMetadataLoader(rc).Load(); err != nil {
+		return err
+	}
+
+	profile := &gitops.Profile{
+		ClusterName: rc.ClusterConfig.Metadata.Name,
+		Path:        o.ProfilePath,
+		CloneFunc:   gitops.CloneContext,
```

Comment:
I am separating all the git operations into it's own tested package perhaps we can inject that here. I'll probably put the PR up on monday.

Reply:
I saw that code is shelling out to git for Git operations (like Flux does), but I'm using a pure Go library `go-git` for clone.

Reply:
Aha, I though there wasn't any library for git operations. In that case, why don't we use the git library for flux? Then we don't need to test the git code, only the integration part.

Reply:
The `go-git` library has not implemented all features that `git` provides (https://github.com/src-d/go-git#comparison-with-git). If all the git operations we're using as part of `install flux` (commit and push) are supported by go-git (I believe it does) then I think we should use it. 
There's also a Go binding for libgit2 but I avoided it because it requires building the libgit2 library and modifying our build process.